File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize){
    int max = 0;
    int minPrice = INT_MAX;
    for (int i = 0; i < pricesSize; i++) {
        if (prices[i] < minPrice) {
            minPrice = prices[i];
         } else if (prices[i] - minPrice > max) {
            max = prices[i] - minPrice;
         }
    }
    return max;    
}

File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        max = 0
        minPrice = sys.maxint
        for i in range(0, len(prices)):
            if prices[i] < minPrice:
                minPrice = prices[i]
            elif prices[i] - minPrice > max:
                max = prices[i] - minPrice
        return max